server/dash/dashboards/nc_summary_comparison.py

```diff
@@ -0,0 +1,754 @@
+import datetime
+import dash_daq as daq
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objects as go
+import requests as re
+
+from config import API_HOST
+from design import DISCRETE_COLORS, LABELS
+from dash import dcc, html, callback
+from dash.dependencies import Input, Output
+from dash.exceptions import PreventUpdate
+
+# Setting 1 week worth of data
+start_date = datetime.date.today() - datetime.timedelta(days=8)
+end_date = datetime.date.today() - datetime.timedelta(days=1)
+
+# Loading the dataframe with the 311 Data API.
+DATE_RANGE_REQ_DATA_API_PATH = f"/requests/updated?start_date={start_date}&end_date={end_date}"
+print(" * Downloading data for dataframe from API path: " + DATE_RANGE_REQ_DATA_API_PATH)
+results = re.get(API_HOST + DATE_RANGE_REQ_DATA_API_PATH)
+data_json = results.json()
+api_data_df = pd.json_normalize(data_json)
```

Comment:
I noticed there are a couple places that are adding `*dateDT` to the df. Maybe it would help to make a function that retrieves the data from the API and does pre-processing that's generally useful (like adding `*dateDT`).

Reply:
Thanks for the suggestion - resolved.